### PR TITLE
Let hazelcast handle session and service ticket counts

### PR DESCRIPTION
**Pull request by mmoayyed**
Allow Hazelcast ticket registry to support session/service ticket count operations. 

Also fixed an issue with the sso sessions report where ops could not be completed due to missing CSRF settings.

this will be selectively ported forward to master.

### CODE REVIEW

File: cas-server-integration-hazelcast/src/main/java/org/jasig/cas/ticket/registry/HazelcastTicketRegistry.java [hunk 1]
```diff
@@ -151,8 +151,32 @@ private void deleteChildren(final TicketGrantingTicket ticket) {
             }
         }
     }
+    
+    @Override
+    public int sessionCount() {
+        int count = 0;
+        final Collection<Ticket> collection = getTickets();
```
Comment (dima767) [inline cas-server-integration-hazelcast/src/main/java/org/jasig/cas/ticket/registry/HazelcastTicketRegistry.java]: I'd do this with J8 stream style, not this imperative style

File: cas-server-integration-hazelcast/src/main/java/org/jasig/cas/ticket/registry/HazelcastTicketRegistry.java [hunk 2]
```diff
@@ -151,8 +151,32 @@ private void deleteChildren(final TicketGrantingTicket ticket) {
...
+            }
+        }
+        return count;
+    }
 
     @Override
+    public int serviceTicketCount() {
+        int count = 0;
+        final Collection<Ticket> collection = getTickets();
+        for (final Ticket ticket : collection) {
```
Comment (dima767) [inline cas-server-integration-hazelcast/src/main/java/org/jasig/cas/ticket/registry/HazelcastTicketRegistry.java]: Same here I.e. stream style

File: cas-server-integration-hazelcast/src/main/java/org/jasig/cas/ticket/registry/HazelcastTicketRegistry.java (hunk 1 above)
Comment (mmoayyed) [inline cas-server-integration-hazelcast/src/main/java/org/jasig/cas/ticket/registry/HazelcastTicketRegistry.java]: I'd love to, except that this is targeted for 4.2 and that branch is based on JDK7. 

Master will be stream-based though.

Comment (dima767) [inline cas-server-integration-hazelcast/src/main/java/org/jasig/cas/ticket/registry/HazelcastTicketRegistry.java]: Ah, got ya. I was also confused about int return type vs long. It's 4.2. OK, nothing to see here. Carry on :-)

**Pull request merged**